src/main/drivers/mcu/stm32/usbd_msc_desc.c

```diff
@@ -33,9 +33,15 @@
 #include "usb_regs.h"
 #include "platform.h"
 
+#if defined (STM32F4)
+#define         DEVICE_ID1          (0x1FFF7A10)
+#define         DEVICE_ID2          (0x1FFF7A14)
+#define         DEVICE_ID3          (0x1FFF7A18)
+#else
 #define         DEVICE_ID1          (0x1FFFF7E8)
 #define         DEVICE_ID2          (0x1FFFF7EA)
 #define         DEVICE_ID3          (0x1FFFF7EC)
+#endif
```

Comment:
I looked into the STM32F4 standard library and found that the enum `UID_BASE` is not defined. This enum might be used in the HAL library.
![image](https://github.com/user-attachments/assets/b9ab59c4-d815-4de7-b54f-c987d3f7ad7f)
![image](https://github.com/user-attachments/assets/07017089-5639-4fae-bc57-52a98739e0e4)

Therefore, should we use Method 1 to directly define the UID values, or Method 2 to indirectly define them through an enum and add the `UID_BASE` enum to the F4 standard peripheral library?

Reply:
oh, stm is so much fun sometimes! imho it would still be better to use `UID_BASE` and just define it ourselves for F4 [some place central](https://github.com/betaflight/betaflight/blob/master/src/main/drivers/mcu/stm32/platform_mcu.h#L101)

Reply:
UID_BASE is defined in CMSIS headers(https://github.com/xpacks/stm32f4-cmsis/blob/xpack/Drivers/CMSIS/Device/ST/STM32F4xx/Include/stm32f405xx.h#L958). 
But it seems that newer version is CMSIS is necessary.

Maintaining local definition is problematic and will lead to problems in future. IMO best way to handle this is to update CMSIS

Reply:
V2.6.0 / 04-November-2016 (https://htmlpreview.github.io/?https://raw.githubusercontent.com/STMicroelectronics/cmsis_device_f4/master/Release_Notes.html) is first version with UID_BASE ...

Reply:
I'll update the CMSIS... and other libraries. Will take a small amount of time as there's some customisation I'll need to bring forward... and I don't have the list.